images/macos/scripts/build/configure-autologin.sh

```diff
@@ -6,9 +6,20 @@
 ##              script was taken from https://github.com/timsutton/osx-vm-templates/blob/master/scripts/autologin.sh
 ################################################################################
 
-echo "Enabling automatic GUI login for the '$USERNAME' user.."
-python3 $HOME/bootstrap/kcpassword.py "$PASSWORD"
-/usr/bin/defaults write /Library/Preferences/com.apple.loginwindow autoLoginUser "$USERNAME"
+source ~/utils/utils.sh
+
+if is_Sonoma; then
+    echo "Enabling autologin for Sonoma"
+    sudo defaults delete /Library/Preferences/com.apple.loginwindow autoLoginUser
+    sudo rm /etc/kcpassword
+    sudo sysadminctl -autologin set -userName "$USERNAME" -password "$PASSWORD"
```

Comment:
`sysadminctl` is available since OS13. For consistency reasons should we switch all images except OS12 to using it?

Reply:
Basically the proposal is
```
ii IsMonterey
  <old behavior>
else
  <use sysadminctl>
end
```

Reply:
Currently we are not using autologin.sh in any images other than macOS12. Will create an internal item to check how it is handled and update that accordingly.